pkg/pool-weighted/contracts/lbp/FixedPriceLBPool.sol

```diff
@@ -0,0 +1,336 @@
+     * non-zero.
+     *
+     * @param exactAmountsInScaled18 The amounts being used to initialize the pool (18-decimal scaled)
+     * @return success Always true: allow the initialization to proceed if all conditions have been met
+     */
+    function onBeforeInitialize(
+        uint256[] memory exactAmountsInScaled18,
+        bytes memory
+    ) public view override onlyBeforeSale returns (bool) {
+        // Verify the sender is the owner through the trusted router
+        if (ISenderGuard(_trustedRouter).getSender() != owner()) {
+            return false;
+        }
+
+        // Validate the initialization amounts are reasonable given the rate
+        _validateInitializationAmounts(exactAmountsInScaled18);
+
+        return true;
+    }
+
+    /**
+     * @notice Validate that initialization amounts are appropriate for the pool configuration.
+     * @dev For one-way pools (blockProjectTokenSwapsIn=true), only project tokens are required since
+     * reserve tokens only flow IN through swaps. For two-way pools, both tokens are required and must
+     * match the expected ratio within tolerance.
+     *
+     * @param amountsScaled18 The scaled initialization amounts
+     */
+    function _validateInitializationAmounts(uint256[] memory amountsScaled18) private view {
+        uint256 projectAmount = amountsScaled18[_projectTokenIndex];
+        uint256 reserveAmount = amountsScaled18[_reserveTokenIndex];
+
+        if (_blockProjectTokenSwapsIn) {
+            // One-way pool: only buying project tokens with reserve.
+            // Therefore, there is no point adding reserve tokens, as they will never be tokenOut in a swap.
+            // This is a form of "seedless" LBP; easy because the math is very simple.
+            if (projectAmount == 0 || reserveAmount != 0) {
+                revert InvalidInitializationAmount();
+            }
+        } else {
+            // Two-way pool: both directions allowed.
+            // Both tokens required and must approximately match the expected ratio.
+            if (projectAmount == 0 || reserveAmount == 0) {
+                revert InvalidInitializationAmount();
+            }
+
+            // Calculate the expected reserve amount based on the project amount and rate.
+            // expectedReserve = projectAmount * rate.
+            uint256 expectedReserve = projectAmount.mulDown(_projectTokenRate);
+
+            // This accounts for decimal precision and gives the owner some flexibility
+            uint256 minReserve = expectedReserve.mulDown(FixedPoint.ONE - _INITIALIZATION_TOLERANCE_PERCENTAGE);
+            uint256 maxReserve = expectedReserve.mulUp(FixedPoint.ONE + _INITIALIZATION_TOLERANCE_PERCENTAGE);
+
+            // Verify the actual reserve amount is within tolerance
+            if (reserveAmount < minReserve || reserveAmount > maxReserve) {
+                revert UnbalancedInitialization();
+            }
```

Comment:
Let's ask about this.

I don't think 2-way pools make sense for fixed price, but if that's the case we don't need to start with reserves.